Fix flaky tests in Datastore sample app tests

```
[ERROR] basicTest(com.example.ApplicationTests)  Time elapsed: 2.1 s  <<< FAILURE!
java.lang.AssertionError: 
[Verify ASC order] 
Expecting:
  <[Singer{singerId='null', firstName='Mary', lastName='Jane', albums=null, firstBand=null, bands=, personalInstruments=},
    Singer{singerId='null', firstName='Scott', lastName='Smith', albums=null, firstBand=null, bands=, personalInstruments=}]>
to contain exactly (and in same order):
  <[Singer{singerId='null', firstName='John', lastName='Doe', albums=null, firstBand=null, bands=, personalInstruments=},
    Singer{singerId='null', firstName='Mary', lastName='Jane', albums=null, firstBand=null, bands=, personalInstruments=},
    Singer{singerId='null', firstName='Scott', lastName='Smith', albums=null, firstBand=null, bands=, personalInstruments=}]>
but could not find the following elements:
  <[Singer{singerId='null', firstName='John', lastName='Doe', albums=null, firstBand=null, bands=, personalInstruments=}]>
```

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/test/java/com/example/ApplicationTests.java
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/test/java/com/example/ApplicationTests.java
@@ -97,6 +97,9 @@ public class ApplicationTests {
 		Singer scottSmith = new Singer(null, "Scott", "Smith", null);
 		Singer frodoBaggins = new Singer(null, "Frodo", "Baggins", null);
 
+		Awaitility.await().atMost(15, TimeUnit.SECONDS)
+				.until(() -> getSingers("/singers?sort=lastName,ASC").size() == 3);
+
 		List<Singer> singersAsc = getSingers("/singers?sort=lastName,ASC");
 		assertThat(singersAsc)
 				.as("Verify ASC order")

